### PR TITLE
Fix alignment for strings containing non-ascii chars

### PR DESCRIPTION
**Pull request by jaunas**
I found an issue when I was trying to align some strings with Polish diacritics. This library counts bytes instead of visible characters, so the more non-ascii characters, the shorter alignment was. I fixed it and added short tests for that.

### CODE REVIEW

**Comment by vitiral** [comment]
Hmm... well bytes is _definitely_ not correct, and I am open to using `chars` due to simplicity.

However, it is more complicated than that if we want "true" width. I'm pretty sure there are characters that are "two widths" and characters that are zero

https://docs.rs/unicode-width/latest/unicode_width/
http://www.unicode.org/reports/tr11/


For example, w̴h̸a̴t̷ ̶i̴s̷ ̸t̴h̴e̷ ̸w̴i̴d̵t̶h̴ ̸o̸f̵ ̴t̷h̸i̸s̵ ̴t̸e̴x̷t̷?̶ (https://lingojam.com/ZalgoText)

**Comment by jaunas** [comment]
Well, indeed it is more complicated than I discovered it myself. I can't see a good simple solution to that problem.
I tested two strings:
- `Ｈｅｌｌｏ, ｗｏｒｌｄ!` from the first link - `chars().count()` fix works the same as rust's `format!`. 
- The blurred text - ZalgoText - spits unicode codes (`w\u{335}\u{302}\u{34e}h\u{334}\u{301}\u{355}a\u{336}` etc.) - I guess it's another problem

File: src/fmtstr.rs [hunk 1]
```diff
@@ -115,13 +115,13 @@ impl<'a, 'b> Formatter<'a, 'b> {
         // precision will limit length
         let len = match precision {
             Some(p) => {
-                if p < s.len() {
+                if p < s.chars().count() {
```
Comment (vitiral) [inline src/fmtstr.rs:118]: Store the result of this once at the top, then use it.

This costs `O(n)` every time it is called, which in some cases here can be at least twice.

File: src/tests/strfmt.rs [hunk 2]
```diff
@@ -62,6 +62,7 @@ fn test_values() {
     vars.insert("x".to_string(), "X".to_string());
     vars.insert("long".to_string(), too_long.clone()); // len=10
     vars.insert("hi".to_string(), "hi".to_string());
+    vars.insert("unicode".to_string(), "ಠ_ಠ".to_string());
```
Comment (vitiral) [inline src/tests/strfmt.rs:65]: can you add a quick tests just validating that `">{unicode:^7}<"` returns the same thing for both `strfmt` and rust's standard `format!` macro?

**Review by vitiral — APPROVED** [review]
If it's the same as rust's implementation then it is good enough for me. A few minor comments. Thanks!

**Comment by jaunas** [comment]
Ok, so I fixed the code as you suggested and now it should be better ;)

**Comment by vitiral** [comment]
Pushing to crates.io as `v0.2.2` now
